CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+# Contributing to Dials
+
+We're happy to consider contributions from outside sources; whether in the form
+of code, tickets, documentation or even just typo corrections! We're aware that
+Dials can at times appear to have relatively indiosyncratic code, and some of
+this is supporting obscure/niche use cases - so, if in the event that you are
+planning any large scale work that you will want to merge back in, please
+contact us beforehand so that we can discuss any potential impacts this might
+have.
+
+Listed in this document are code standards and conventions that you should try
+to adhere to, some of which are essential and others that are just encouraged.
+The intention is that all of the code should try to converge towards these.
+
+## Configuring Dials for development
+
+1.  Install Dials and it's cctbx_project dependencies for development; For
+    Linux/macOS, the current best way to create a fresh installation of Dials
+    and all of it's dependencies is with the following commands:
+    ```
+    git clone https://github.com/cctbx/cctbx_project modules/cctbx_project
+    ln -s modules/cctbx_project/libtbx/auto_build/bootstrap.py
+    python bootstrap.py hot base update build --builder=dials --use-conda
+    ```
+2.  Activate the environment with `source <root>/build/setpaths.sh`. This will
+    need to be done every time you work on dials code.
+3.  Install pre-commit hooks with `libtbx.precommit install`. This will install
+    hooks for Dials, dxtbx and xia2.
+
+The Dials repository is now checked out in `<root>/modules/dials`. During
+development, run tests with pytest with `libtbx.pytest --regression` to ensure
+that all of the tests still pass.
+
+If you update or change dependencies, you may occasionally need to regenerate
+the static libtbx ecosystem by running `libtbx.refresh` - this will also
+generate an alias to any executable scripts installed by `pip` as
+`libtbx.<script>`. Using these helps ensure that you are using the bundled
+python version instead of your system version.
+
+## Essential Code Development Standards
+
+- **Please install the pre-commit hooks**. (`libtbx.precommit install` if using
```

Comment:
Does this really need to be essential? I have found pre-commit to be very flaky on my machine. I.e. it has never worked for me with the DIALS repository.

Reply:
Yes. (And if it doesn't work then that is a bug that should be reported please.)

Reply:
We just had this discussion - turns out @jmp1985 had the *old* precommit hook/tool installed. With the new-style it works fine!

Reply:
Also, while maybe ontologicallly confusing, this entry is a "please" whilst the running of black and flake8 is a separate entry in essential and also "try to" because if you really can't/don't want to install precommit/doing an essential fix on an unfamiliar computer, then running them separately is fine, but if you don't, expect your code to be changed by someone else (or the automatic tools) downstream to be compliant.

As another example, if you commit a 100mb file because you didn't have the precommit installed, expect @Anthchirp to frown at you and your commit to be reverted.